NoQuD/nodal/Nodal.py

```diff
@@ -78,12 +78,48 @@ def current_interface_condition(self):
                     rhs2 = pow(-1, k) * f[i][2 * j + 1]
                     self.linear_system[i][self.nodes + 1 + j][self.order_of_legendre_poly * (j + 1) + k] = rhs2
 
+    def balance_condition(self):
+
+        for i in xrange(0, self.groups):
```

Comment:
I realize this may be a lot of work, but it might be worth going through and renaming all of your iterations over i,j,k to iterations over the actual variable you are iterating over. For example., if 'i' is the x-direction potentially change the iterating variable from 'i' to 'x' to make the code more clear.